lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -563,13 +569,38 @@ def _run_script(self, rerun_data: RerunData) -> None:
 
         except BaseException as e:
             self._session_state[SCRIPT_RUN_WITHOUT_ERRORS_KEY] = False
-            handle_uncaught_app_exception(e)
+            uncaught_exception = e
```

Comment:
Is there a reason for changing this?

Reply:
Yep, `uncaught_exception` is saved here to be added to the page profile in case it had an exception.

Reply:
Ah I see. I didn't catch that for some reason.